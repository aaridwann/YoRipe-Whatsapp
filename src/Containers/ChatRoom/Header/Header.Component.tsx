import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

import Constants from '../../../Constants/Constants';
import { scaleSize } from '../../../Utils/Size/size.utils';
import { HooksInterface, ListData, Props } from '../ListChat.Types';
import { styles } from './Header.Component.Styles';

/**
 *
 * @param {HooksInterface} hooks
 * @returns {React.ReactNode}
 */
const _renderCallIcon = (hooks: HooksInterface): React.ReactNode => (
    <TouchableOpacity style={styles.icon} onPress={() => hooks.onCallHandler}>
        <Ionicons
            name="call"
            size={scaleSize(18)}
            color={Constants.Colors.WHITE}
        />
    </TouchableOpacity>
);

/**
 *
 * @param {HooksInterface} hooks
 * @returns {React.ReactNode}
 */
const _renderVideoIcon = (hooks: HooksInterface): React.ReactNode => (
    <TouchableOpacity style={styles.icon} onPress={() => hooks.onVideoHandler}>
        <Ionicons
            name="videocam"
            size={scaleSize(18)}
            color={Constants.Colors.WHITE}
        />
    </TouchableOpacity>
);

/**
 * Render Name for Left header Props for Top bar navigation
 * @param {ListData} params - Params of chat room
 * @returns {React.ReactNode} - Render name Render Header Left TopBarNavigation
 */
const _renderName = (params: ListData): React.ReactNode => (
    <View>
        <Text style={styles.name}>{params.name}</Text>
        <Text style={styles.lastSeen}>Last seen today 11:00 AM</Text>
    </View>
);

/**
 * Render Profile Left header Props for Top bar navigation
 * @param {HooksInterface} hooks - Hooks of Chat room
 * @param {Props} props - Props of chat room
 * @returns {React.ReactNode} - Render Profile Render Header Left TopBarNavigation
 */
const _renderProfile = (
    hooks: HooksInterface,
    props: Props
): React.ReactNode => {
    const { params } = props.route;

    return (
        <TouchableOpacity
            style={styles.profile}
            onPress={hooks.onProfileHandler}
        >
            <Image style={styles.avatar} source={{ uri: params.image }} />
            {_renderName(params)}
        </TouchableOpacity>
    );
};

/**
 *
 * @param {HooksInterface} hooks - Hooks Chatroom
 * @returns {React.ReactNode}
 */
const RightHeaderChatRoom = (hooks: HooksInterface): React.ReactNode => (
    <View style={{ flexDirection: 'row' }}>
        {_renderCallIcon(hooks)}
        {_renderVideoIcon(hooks)}
    </View>
);

/**
 * Render Left header for chat room page
 *
 * @param {HooksInterface} hooks - Hooke of chat room
 * @param {Props} props - Props of chat room
 * @returns
 */
const LeftHeaderChatRoom = (
    hooks: HooksInterface,
    props: Props
): React.ReactNode => (
    <View style={styles.leftWrapper}>{_renderProfile(hooks, props)}</View>
);

export { RightHeaderChatRoom, LeftHeaderChatRoom };
