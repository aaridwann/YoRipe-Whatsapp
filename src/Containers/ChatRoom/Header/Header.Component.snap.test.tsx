import React from 'react';
import { render } from '@testing-library/react-native';

import { LeftHeaderChatRoom, RightHeaderChatRoom } from './Header.Component';
import { HooksInterface } from '../ListChat.Types';

beforeEach(() => {
    jest.clearAllMocks();
});

jest.mock('@expo/vector-icons/Ionicons', () => 'Icons');

describe('Header Component', () => {
    const listData = {
        check: false,
        id: 0,
        text: '',
        time: '',
        type: 'author',
    };

    const mockState = {
        isLoading: false,
        input: '',
        isOpenOptionsId: null,
        markChat: [],
        image: '',
        isShowSearch: false,
        chatData: [listData],
        chatFilter: [listData],
        uploadImgUri: '',
    };

    const mockHooks: HooksInterface = {
        state: { ...mockState },
        infoHandler: jest.fn(),
        clearInfo: jest.fn(),
        inputHandler: jest.fn(),
        submitHandler: jest.fn(),
        attachHandler: jest.fn(),
        showSearchToggle: jest.fn(),
        backHandler: jest.fn(),
        searchMessageHandler: jest.fn(),
        onProfileHandler: jest.fn(),
        discardImg: jest.fn(),
        onCallHandler: jest.fn(),
        onVideoHandler: jest.fn(),
    };

    const ListData = {
        name: '',
        image: '',
        latestChat: '',
        date: '',
        notification: '',
        phone: '',
    };

    const mockProps = {
        navigation: {
            goBack: jest.fn(),
            push: jest.fn(),
        },
        route: {
            params: ListData,
        },
    };

    describe('LeftHeaderComponent', () => {
        it('Should render with correct value', () => {
            expect(
                render(LeftHeaderChatRoom(mockHooks, mockProps))
            ).toMatchSnapshot();
        });
    });

    describe('RightHeaderComponent', () => {
        it('Should render with correct value', () => {
            const Component = () => RightHeaderChatRoom(mockHooks);

            expect(render(<Component />)).toMatchSnapshot();
        });
    });
});
