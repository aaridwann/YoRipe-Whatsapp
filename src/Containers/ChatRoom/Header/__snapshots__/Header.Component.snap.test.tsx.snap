// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component LeftHeaderComponent Should render with correct value 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "borderRadius": 30,
          "height": 60,
          "marginHorizontal": 8,
          "width": 60,
        }
      }
    />
    <View>
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 30,
            "fontWeight": "500",
          }
        }
      />
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
            "opacity": 0.5,
          }
        }
      >
        Last seen today 11:00 AM
      </Text>
    </View>
  </View>
</View>
`;

exports[`Header Component RightHeaderComponent Should render with correct value 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <Icons
      color="#ffffff"
      name="call"
      size={36}
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginHorizontal": 8,
        "opacity": 1,
      }
    }
  >
    <Icons
      color="#ffffff"
      name="videocam"
      size={36}
    />
  </View>
</View>
`;
